sound/soc/sof/ipc4-topology.c

```diff
@@ -197,6 +197,32 @@ static void sof_ipc4_dbg_audio_format(struct device *dev,
 	}
 }
 
+static const struct sof_ipc4_audio_format *
```

Comment:
commit message: what happens if there is a non-process module between copier and process?

It's not uncommon to have e.g. gain control before and after a process.

how would the copier output pin be set then?

Reply:
thats exactly what line 206 below does

Reply:
so how does this work between the non-process module and the process module? 

I guess I am totally missing how the format propagation and negotiation works. It's not like I've seen any explanations though.